### DISCUSSION TITLE
Saving TTFont as UFO

### DISCUSSION THREAD

**Discussion by NightFurySL2001** [fonttools/fonttools#3196] [Q&A] [answered]
Is it possible to save a `TTFont` to UFO format purely with FontTools `ufoLib`? I'm looking for this using it with fontBuilder.

**Comment by justvanrossum** [comment, marked as answer]
Sure, just not with a single line of code. So, no, actually. The "extractor" library can do much of the heavy lifting: https://github.com/robotools/extractor

**Reply by NightFurySL2001** [reply]
That's true, but I have some problems with `extractor`:

1. There are no explicit function that allows TTFont to UFO. The main code only accept file name as input and then convert the file to TTFont in the function. I solved this by importing the four functions I needed.
2. I don't really need the OpenType features section (which I added, yes....) but importing the whole `fontFeatures` library costs like an extra 5-10MB to the final software. I can remove it in my venv but that's not really solving it.
3. `extractor` raises two error: `AttributeError: 'dict' object has no attribute 'bFamilyType'` and  
```
versionMajor, versionMinor = version.split(".")
ValueError: not enough values to unpack (expected 2, got 1)
```
The later is caused by me setting the version number to an integer and not a float: `fb.setupHead(fontRevision=headVersion)`.
I can't catch the bug for the former. Maybe it's due to `fontbuilder` doesn't build panose?

**Reply by moyogo** [reply]
1. Is a feature request or PR for extractor, the extract functions could handle being given a TTFont object.
2. As well, this can be handled with optional dependencies and conditional imports.
3. Sounds like a bug in extractor or your code.

Either way, report these on the extractor repository, they sound reasonable.

**Comment by moyogo** [comment]
There's https://github.com/robotools/extractor for that, but you already know about it.
